test/StakeManager.js

```diff
@@ -640,23 +704,29 @@ describe('StakeManager', function () {
 
     it('Delegator should not be able to withdraw if didnt unstake', async function () {
       const stakerId = await stakeManager.stakerIds(signers[1].address);
-      const tx = stakeManager.connect(signers[5]).withdraw(stakerId);
+      const tx = stakeManager.connect(signers[5]).initiateWithdraw(stakerId);
       await assertRevert(tx, 'Did not unstake');
     });
 
     it('Delegator should be able to unstake when there is no existing lock', async function () {
       await mineToNextEpoch();
-      const epoch = await getEpoch();
+      let epoch = await getEpoch();
       const amount = tokenAmount('10000'); // unstaking partial amount
       let staker = await stakeManager.getStaker(4);
-      const prevStake = staker.stake;
       const sToken = await stakedToken.attach(staker.tokenAddress);
-      const totalSupply = await sToken.totalSupply();
+      await sToken.connect(signers[5]).approve(stakeManager.address, amount);
       await stakeManager.connect(signers[5]).unstake(staker.id, amount);
-      const lock = await stakeManager.locks(signers[5].address, staker.tokenAddress);
+      for (let i = 0; i < UNSTAKE_LOCK_PERIOD; i++) {
+        await mineToNextEpoch();
+      }
+      epoch = await getEpoch();
+      const prevStake = staker.stake;
+      const totalSupply = await sToken.totalSupply();
       const rAmount = (amount.mul(staker.stake)).div(totalSupply);
+      await stakeManager.connect(signers[5]).initiateWithdraw(staker.id);
       staker = await stakeManager.getStaker(4);
 
+      const lock = await stakeManager.locks(signers[5].address, staker.tokenAddress, 1);
       const initial = await sToken.getRZRDeposited(signers[5].address, amount); // How much delegator had put for this much amount of SRZRS
       const gain = (rAmount.sub(initial)); // commission in accordance to gain
       const commission = ((gain).mul(staker.commission)).div(100);
```

Comment:
hmm I think this is the reason why tests not failed with this change
we are calling contract to getRazorDeposited 
so its in line with contract but not logically

Reply:
Yep, was just about to say the same thing.
I'll look more into this

Reply:
resolved?

Reply:
Yep resolved